nexus/types/src/deployment.rs

```diff
@@ -605,6 +691,150 @@ pub struct OperatorNexusConfig<'a> {
     pub external_dns_servers: &'a [IpAddr],
 }
 
+/// Argument to [`Blueprint::expunged_zones()`] allowing the caller whether they
+/// zones that are ready for cleanup, not ready for cleanup, or both/either.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ReadyForCleanup {
+    /// Only return zones that are ready for cleanup (i.e., have been confirmed
+    /// shut down by sled-agent and will not be restarted).
+    Yes,
+    /// Only return zones that are NOT YET ready for cleanup. These zones are
+    /// expunged but could still be running.
+    No,
+    /// Return expunged zones regardless in either "ready for cleanup" state.
+    Both,
+}
+
+/// Argument to [`Blueprint::expunged_zones()`] requiring the caller to specify
+/// the reason they want to access a blueprint's expunged zones.
+///
+/// The planner is responsible for permanently pruning expunged zones from the
+/// blueprint. However, doing so correctly requires waiting for any cleanup
+/// actions that need to happen, all of which are specific to the zone kind.
+/// For example, blueprint execution checks for expunged Oximeter zones to
+/// perform metric producer reassignment, which means the planner cannot prune
+/// an expunged Oximeter zone until it knows that all possible such
+/// reassignments are complete.
+///
+/// These reasons are not used at runtime; they exist soley to document and
+/// attempt to statically guard against new code adding a new call to
+/// `expunged_zones()` that the planner doesn't know about (and therefore
+/// doesn't get updated to account for!). If you are attempting to call
+/// `expunged_zones()` for a new reason, you must:
+///
+/// 1. Add a new variant to this enum.
+/// 2. Update the planner to account for it, to prevent the planner from pruning
+///    the zone before whatever your use of it is completes.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum BlueprintExpungedZoneAccessReason {
```

Comment:
I assume there is going to be a match on all of these in the planner when we work towards closing https://github.com/oxidecomputer/omicron/issues/5552 , so that "if you add a new access-reason, you'll also be forced to do the accounting in the planner", right?

Seems like it might be easy for this to get out-of-sync if this enum is an "unused arg from callers" that never actually has any compile-time connection to the planner work